frontend/src/components/search/summary.js

```diff
@@ -0,0 +1,139 @@
+import React, {Component} from 'react';
+import {OutClickable} from "../dropdown";
+import connect from "react-redux/es/connect/connect";
+
+class SummaryDropdown extends OutClickable {
+    constructor(props) {
+        super(props);
+        this.collapse = this.collapse.bind(this);
+        this.toggle_dropdown = this.toggle_dropdown.bind(this);
+        this.computeStats = this.computeStats.bind(this);
+        this.parseTime = this.parseTime.bind(this);
+        this.state = {active: false};
+    }
+
+    collapse() {
+        this.setState({active: false});
+    }
+
+    activate_dropdown() {
+        this.setState({active: true});
+    }
+
+    toggle_dropdown() {
+        if (this.state.active) {
+            this.collapse();
+        } else {
+            this.activate_dropdown();
+        }
+    }
+
+    computeStats() {
+        // Computes the following statistics from schedule from props
+        // 1. Average hours per day
+        // 2. Average difficulty 
+        // 3. Earliest start time
+        // 4. Latest end time 
+        // 5. Max hours a day
+
+        let data = {
+            "avghr" : 0,
+            "avgdf" : 0,
+            "earlyt" : 100, 
+            "latet" : -1, 
+            "maxhoursd" : 0
+        };
+
+        if (!this.props.schedData) {
+            return data;
+        }
+        
+        let courseList = this.props.schedData.meetings;
+        let numCourses = courseList.length;
+        let dayHoursMap = new Map();
+
+        courseList.map(course => {
+            data.earlyt = Math.min(data.earlyt, course.meetHour);
+            data.latet = Math.max(data.latet, course.meetHour + course.hourLength);
+            data.avgdf += course.revs.cD;
+            let meetDay = course.meetDay;
+            [...meetDay].forEach(c => {
+                let currNum = dayHoursMap.get(c) ? dayHoursMap.get(c) : 0;
+                dayHoursMap.set(c, currNum + course.hourLength);
+            })
+        });
+
+        for (let [k,v] of dayHoursMap) {
+            data.maxhoursd = Math.max(data.maxhoursd, v);
+            data.avghr += v;
+        }
+
+        data.avghr /= 5;
+        data.avgdf /= numCourses;
+
+
+        return data;
+    }
+
+    parseTime(time) {
+        if (time === "N/A") {
+            return time;
+        }
+        if (time > 11) {
+            if (time > 12) {
+                return (time - 12) + " PM";
+            } 
+            return time + " PM";
+        } else {
+            return time + " AM";
+        }
+    } 
+
+    render() {
```

Comment:
It seems like a lot of this code is duplicating things in the `Dropdown` component found at https://github.com/pennlabs/PennCoursePlan/blob/master/frontend/src/components/dropdown.js. We should try and reuse components wherever possible. Instead of duplicating the render logic for the `Dropdown` component, try to use the existing `Dropdown` component to render the schedule stats. An example of `Dropdown` being used can be found at [line 49 of the searchbar component](https://github.com/pennlabs/PennCoursePlan/blob/master/frontend/src/components/search/bar.js#L49), and you can use it as a guide.